### PR TITLE
Use new filter in global search

### PR DESCRIPTION
**Pull request by SferaDev**
Use the new search branch filters

### CODE REVIEW

File: test/integration/search.test.ts [hunk 1]
```diff
@@ -166,6 +166,21 @@ describe('search', () => {
...
     }
   });
+
+  test.skip('search all with filters', async () => {
+    const results = await client.search.all('fruits', {
+      filter: { users: { full_name: 'Team fruits' } }
+    });
+
+    for (const result of results) {
+      expect(result.table).toBe('users');
```
Comment (SferaDev) [inline test/integration/search.test.ts:176]: Need to review this after the backend team finishes their discussion

https://github.com/xataio/xata/pull/825#issuecomment-1162199983

File: packages/client/src/search/index.ts [hunk 2]
```diff
@@ -2,14 +2,22 @@ import { getBranchDetails, searchBranch } from '../api';
...
 import { SelectedPick } from '../schema/selection';
-import { GetArrayInnerType, StringKeys, Values } from '../util/types';
+import { GetArrayInnerType, StringKeys, UnionToIntersection, Values } from '../util/types';
 
 export type SearchOptions<Schemas extends Record<string, BaseData>, Tables extends StringKeys<Schemas>> = {
   fuzziness?: number;
   tables?: Tables[];
+  filter?: UnionToIntersection<
+    Values<{
+      [Model in GetArrayInnerType<NonNullable<NonNullable<Tables[]>>>]: {
```
Comment (SferaDev) [inline packages/client/src/search/index.ts:16]:
```suggestion
      [Model in GetArrayInnerType<NonNullable<Tables[]>>]: {
```

**Review by gimenete — APPROVED** [review]
🚀

File: .changeset/spotty-monkeys-rhyme.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@xata.io/client': patch
```
Comment (SferaDev) [inline .changeset/spotty-monkeys-rhyme.md:2]:
```suggestion
'@xata.io/client': minor
```

**Pull request merged**